src/cli/self_update/windows.rs

```diff
@@ -179,14 +179,14 @@ pub(crate) async fn try_install_msvc(opts: &InstallOpts<'_>) -> Result<ContinueI
         .context("error creating temp directory")?;
 
     let visual_studio = tempdir.path().join("vs_setup.exe");
-    let download_tracker = DownloadTracker::new_with_display_progress(true);
-    download_tracker.lock().unwrap().download_finished();
+    let mut download_tracker = DownloadTracker::new_with_display_progress(true);
```

Comment:
I guess we might need to return to the original semantics (`Arc<Mutex<T>>`) in this particular case: https://stackoverflow.com/a/30560208

Reply:
Not exactly sure what that code is doing, it looks fishy...